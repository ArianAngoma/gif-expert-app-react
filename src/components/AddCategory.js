import {useState} from "react";
import PropTypes from "prop-types";

const AddCategory = ({setCategories}) => {
    const [inputValue, setInputValue] = useState('');

    // Handle para poder cambiar el estado de inputValue
    const handleInputChange = (e) => {
        setInputValue(e.target.value)
    }

    const handleSubmit = (e) => {
        // Previene el evento por default del formulario, en este caso que no recargue la pÃ¡gina al enviar el formulario
        e.preventDefault();

        // Llamar la propiedad setCategories del componente padre GifExpertApp
        if (inputValue.trim().length > 2) {
            setCategories(cats => [...cats, inputValue]);

            // Limpiar el input al enviar el formulario
            setInputValue('');
        }
    }

    return (
        <form onSubmit={handleSubmit}>
            <input
                type="text"
                value={inputValue}
                onChange={handleInputChange}
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}

export default AddCategory;