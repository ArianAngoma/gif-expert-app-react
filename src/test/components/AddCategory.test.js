import {shallow} from "enzyme";
import AddCategory from "../../components/AddCategory";

describe('Test to component <AddCategory/>', () => {
    // Funcion de jest
    const setCategories = jest.fn();

    let wrapper = shallow(<AddCategory setCategories={setCategories}/>);

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories}/>);
    });

    test('Should display the component <AddCategory/> correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Should change the text box', () => {
        const input = wrapper.find('input');

        // Valor que se envía a la caja de texto
        const value = 'Hola Mundo';

        // Simular cambio en la caja de texto
        input.simulate('change', {target: {value}});
    });

    test('Should not send the form', () => {
        wrapper.find('form').simulate('submit', {
            preventDefault() {
            }
        });

        // setCategories no se debería de haber llamado porque no se envía nada el input text
        expect(setCategories).not.toHaveBeenCalled();
    })
})