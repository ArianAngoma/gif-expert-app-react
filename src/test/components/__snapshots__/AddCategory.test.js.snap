// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test to component <AddCategory/> Should display the component <AddCategory/> correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
